### DISCUSSION TITLE
Is it possible to nest unicorn views?

### DISCUSSION THREAD

**Discussion by Maxtrix-dev** [django-commons/django-unicorn#615] [Q&A]
I have a problem with nesting unicorn views.
My `Layout.html` (the first unicorn view) looks like this (this si the only important part this unicorn view works fine):
```html
{% elif pageIndex == 2 %}
   {% unicorn 'Modules' %}
``` 
And my `Modules.html` looks like this (this is the second unicorn view where the unicorn:click doesn't work):
```html
{% load unicorn %}
<div>
        {% unicorn_scripts %}
        {{modules}}
        <button u:click="testThis()">test</button>
</div>
```
and my `Modules.py` looks like this:
```html
from django_unicorn.components import UnicornView


class ModulesView(UnicornView):
    modules="SRT"
    def testThis(self):
        print("test This")
        
    def mount(self):
        print("mounted")
        return super().mount()
```

It mounts but the click doesn't work do you anyone have any suggestion / author should it even work?

**Comment by adamghill** [comment]
Hi, I assume you have you looked through [these docs](https://www.django-unicorn.com/docs/child-components/)? There are definitely some "gotchas" with nesting components, but they should work. Some example code of nested components is here: https://github.com/adamghill/django-unicorn/tree/main/example/unicorn/templates/unicorn/nested. If you create a repo that replicates your issue with sample code that I can run, I can probably help get something working. 👍
